design-docs/done/build-initialisation.md

```diff
@@ -166,3 +166,14 @@ This story adds a commandline property `--gradle-version` to the `wrapper` task
 
 * Running `gradle wrapper --gradle-version 1.6` generates valid `wrapper.properties` with correct URL.
 
+# Story: User specifies target gradle distribution type when generating the wrapper
+
+This story adds a command line option called `--gradle-type` to the `wrapper` task to specify the desired Gradle distribution type:
+
+* Add `--gradle-type` command-line option to `wrapper`.
+
+Being able to set the distibution type to `all` instead of the default of `bin` is useful if the source code is required for IDEs to offer proper auto-completion for the DSL.
+
+## Test coverage
+
+* Running `gradle wrapper --gradle-version 2.13 --gradle-type all` generates valid `wrapper.properties` with the expected URL.
```

Comment:
There should be more test coverage:

 - What happens if there is no `--dist-type` command line option
 - What happens if the `--dist-type` in the command line does not exist